Flush pending callbacks

grgustaf opened the pull request.

grgustaf commented:
Note, reordering zjs_callback_t took it from 48 bytes to 40 bytes (on Linux at least).

jprestwo commented on src/zjs_callbacks.c:
```diff
@@ -323,17 +343,33 @@ void zjs_remove_callback(zjs_callback_id id)
+                                            CB_FLUSH_ONE, NULL, 0);
+            if (ret) {
+                // couldn't add flush command, so just free now
+                zjs_flush_callback(id);
```
Maybe print an error/debug message here because this likely means the ring buffer is full.

grgustaf commented:
OK, I added a debug print and also renamed flush function to free because "flush" really means ensuring the pending callbacks get flushed out of the queue safely, not the final freeing of the callback afterward.

jprestwo commented:
LGTM assuming travis is happy.

jimmy-huang commented:
+1
